bindMount creates any leading missing dirs

tobsan opened the pull request:
This goes for both files and directories.

Signed-off-by: Tobias Olausson <tobias.olausson@pelagicore.com>

Commit:
bindMount creates any leading missing dirs

This goes for both files and directories.

Signed-off-by: Tobias Olausson <tobias.olausson@pelagicore.com>

DunderRoffe reviewed (approved):
Overall I think this looks  good. However, you could merge the multiple depth folder test with the generic folder mounting test. Having them separate does not add that much value.

Commit:
Fixes test failure with folder mounting

The test assumed a char pointer was untouched, but the previous
test modified it with strcat.

Signed-off-by: Tobias Olausson <tobias.olausson@pelagicore.com>

tobsan commented:
@DunderRoffe since they are handled slightly differently in the code it does make sense to test both of them imho.
